x-pack/plugins/security_solution/public/entity_analytics/components/risk_score_preview_section.tsx

```diff
@@ -0,0 +1,192 @@
+}
+
+const getRiskiestScores = (scores: RiskScore[] = [], field: string) =>
+  scores
+    ?.filter((item) => item?.identifierField === field)
+    ?.sort((a, b) => b?.totalScoreNormalized - a?.totalScoreNormalized)
+    ?.slice(0, 5) || [];
+
+const RiskScorePreviewPanel = ({
+  items,
+  showMessage,
+  hideMessage,
+  isLoading,
+  type,
+}: IRiskScorePreviewPanel) => {
+  const [trigger, setTrigger] = useState<'closed' | 'open'>('open');
+  const onToggle = (isOpen: boolean) => {
+    const newState = isOpen ? 'open' : 'closed';
+    setTrigger(newState);
+  };
+
+  return (
+    <EuiPanel hasBorder={true}>
+      <EuiAccordion
+        initialIsOpen={true}
+        isLoading={isLoading}
+        id={'host-table'}
+        buttonContent={trigger === 'closed' ? showMessage : hideMessage}
+        forceState={trigger}
+        onToggle={onToggle}
+        extraAction={<EuiIcon type={type === RiskScoreEntity.host ? 'storage' : 'user'} />}
+      >
+        <>
+          <EuiSpacer size={'m'} />
+          <RiskScorePreviewTable items={items} type={type} />
+        </>
+      </EuiAccordion>
+    </EuiPanel>
+  );
+};
+
+export const RiskScorePreviewSection = () => {
+  const [dateRange, setDateRange] = useState<{ from: string; to: string }>({
+    from: 'now-24h',
+    to: 'now',
+  });
+
+  const [filters, setFilters] = useState<{ bool: BoolQuery }>({
+    bool: { must: [], filter: [], should: [], must_not: [] },
+  });
+
+  const {
+    unifiedSearch: {
+      ui: { SearchBar },
+    },
+  } = useKibana().services;
+
+  const { addError } = useAppToasts();
+
+  const { data, isLoading, refetch, isError } = useRiskScorePreview({
+    filter: filters,
+    range: {
+      start: dateRange.from,
+      end: dateRange.to,
+    },
+  });
+
+  const { indexPattern } = useSourcererDataView(SourcererScopeName.detections);
+
+  const hosts = getRiskiestScores(data?.scores, 'host.name');
+  const users = getRiskiestScores(data?.scores, 'user.name');
+
+  if (isError) {
+    return (
+      <EuiCallOut
+        data-test-subj="risk-preview-error"
+        title={i18n.PREVIEW_ERROR_TITLE}
+        color="danger"
+        iconType="error"
+      >
+        <p>{i18n.PREVIEW_ERROR_MESSAGE}</p>
+        <EuiButton
+          data-test-subj="risk-preview-error-button"
+          color="danger"
+          onClick={() => refetch()}
+        >
+          {i18n.PREVIEW_ERROR_TRY_AGAIN}
+        </EuiButton>
+      </EuiCallOut>
+    );
+  }
+
+  return (
+    <>
+      <EuiTitle>
+        <h2>{i18n.PREVIEW}</h2>
+      </EuiTitle>
+      <EuiSpacer size={'s'} />
+      <EuiText>{i18n.PREVIEW_DESCRIPTION}</EuiText>
+      <EuiSpacer />
+      <EuiFormRow fullWidth>
+        {indexPattern && (
+          <SearchBar
+            appName="siem"
+            isLoading={isLoading}
+            indexPatterns={[indexPattern] as DataView[]}
+            dateRangeFrom={dateRange.from}
+            dateRangeTo={dateRange.to}
+            onQuerySubmit={(payload) => {
```

Comment:
Should this callback be memoized? We might be causing SearchBar to rerender unnecessarily.